optional_plugins/resultsdb/avocado_resultsdb/__init__.py

```diff
@@ -0,0 +1,172 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2017
+# Authors: Amador Pahim <apahim@redhat.com>
+
+"""
+Avocado Plugin to propagate Job results to Resultsdb
+"""
+
+import os
+import sys
+import time
+import resultsdb_api
+
+from avocado.core.plugin_interfaces import CLI, ResultEvents
+from avocado.core.settings import settings
+from avocado.core import exceptions
+from avocado.utils import stacktrace
+
+
+class ResultsdbResult(ResultEvents):
+
+    name = 'resultsdb'
+    description = 'Resultsdb result support'
+
+    def __init__(self, args):
+
+        self.rdbapi = None
+        if getattr(args, 'resultsdb_api', None) is not None:
+            self.rdbapi = resultsdb_api.ResultsDBapi(args.resultsdb_api)
+
+        self.rdblogs = None
+        if getattr(args, 'resultsdb_logs', None) is not None:
+            self.rdblogs = args.resultsdb_logs
+
+        self.job_id = None
+        self.job_logdir = None
+
+    def pre_tests(self, job):
+        if self.rdbapi is None:
+            return
+
+        self.job_id = job.unique_id
+        self.job_logdir = os.path.basename(job.logdir)
+
+        ref_url = None
+        if self.rdblogs is not None:
+            ref_url = '%s/%s' % (self.rdblogs, self.job_logdir)
+
+        self.rdbapi.create_group(self.job_id, ref_url)
+
+    def start_test(self, result, state):
+        pass
+
+    def end_test(self, result, state):
+        if self.rdbapi is None:
+            return
+
+        status = state['status']
+        outcome = self._status_map(status)
+
+        name = state['name'].name
+
+        group = [self.job_id]
+
+        note = None
+        if state['fail_reason'] is not None:
+            note = str(state['fail_reason'])
+
+        ref_url = None
+        if self.rdblogs is not None:
+            logdir = os.path.basename(state['logdir'])
+            ref_url = '%s/%s/test-results/%s' % (self.rdblogs,
+                                                 self.job_logdir,
+                                                 logdir)
```

Comment:
OK, so the `rdblogs` will be appended with `$job_logdir/test-results/$logdir`. This is quite important information and it'd be nice to have it in the documentation. (I was wondering how resultsdb assigns logs based on the id or so...)